File: FizzBuzz.cpp
// Runtime: 3ms - Beats 90.66%
// Memory: 7.1MB - Beats 96.95%
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> numList(n);
        for (int i = 1; i <= n; i++){
            if (i % 15 == 0)
                numList[i - 1] = ("FizzBuzz");
            else if (i % 3 == 0)
                numList[i - 1] = ("Fizz");
            else if (i % 5 == 0)
                numList[i - 1] = ("Buzz");
            else
                numList[i - 1] = (to_string(i));
        }
        return numList;
    }
};

File: FizzBuzz.py
# Runtime: 37ms - Beats 98.79%
# Memory: 15MB - Beats 86.31%
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        def convert(n: int) -> str:
            if n % 3 == 0 and n % 5 == 0:
                return "FizzBuzz"
            elif n % 3 == 0:
                return "Fizz"
            elif n % 5 == 0:
                return "Buzz"
            else:
                return str(n)

        return [convert(num) for num in range(1, n + 1)]
